ntbs-service/Pages/Shared/_ClusterInformation.cshtml

```diff
@@ -2,13 +2,26 @@
 @inject IReportingLinksService ReportingLinksService
 @model ntbs_service.Models.Entities.Notification
 
+@{
+    var clusterLink = ReportingLinksService.GetClusterReport(Model.ClusterId);
+}
+
 @if (Model.ClusterId != null)
 {
     <h3>Cluster Information</h3>
     <div>
         <p>
             <strong>@Html.DisplayNameFor(m => m.ClusterId): </strong>
-            <a href="@(ReportingLinksService.GetClusterReport(Model.ClusterId))"> @Model.ClusterId </a>
+            @if (string.IsNullOrEmpty(clusterLink))
+            {
+                @Model.ClusterId
+            }
+            else
+            {
+                <a href="@clusterLink" target="_blank" rel="noopener noreferrer">
+                    @Model.ClusterId
+                </a>
+            }

```

Comment:
Is this similar enough to the one above that we can use a partial?

Reply:
Yeah we can. I've added a `_ClusterIdLink` partial to do this. Due to where the two existing partials are being used we can't refactor them out, but at least the code is now shared.